applications/GeoMechanicsApplication/custom_strategies/strategies/geo_mechanics_newton_raphson_erosion_process_strategy.hpp

```diff
@@ -239,9 +219,10 @@ class GeoMechanicsNewtonRaphsonErosionProcessStrategy
     /// </summary>
     /// <param name="PipeElements"></param>
     /// <returns></returns>
-    int InitialiseNumActivePipeElements(std::vector<Element*> PipeElements)
+    template <typename FilteredElementType>
```

Comment:
Very nice templated solution to handle both types of elements. You could consider renaming the type a bit to make it more clear that its a vector of elements, for example:
```suggestion
    template <typename FilteredElementVectorType>
```
or
```suggestion
    template <typename FilteredElementsType>
```

Reply:
choose the latter. FilteredElementsType